### PR TITLE
optimized : handle non-runtime exp in exceptionHandleTemplate()

### PR DESCRIPTION
**Pull request by Bughue**
- [ ] I have registered the PR [changes](https://github.com/seata/seata/tree/develop/changes).

### Ⅰ. Describe what this PR did
现在exceptionHandleTemplate没有处理非运行时的异常，当异常发生时，这里没有日志，也没有handle 。建议是至少要有日志，onException()可以另外讨论。
我们的场景是数据库响应慢，druid抛出了异常，但是handler没处理，被通信组件直接获取异常message [给回TM了]

### Ⅱ. Does this pull request fix one issue?

fixed https://github.com/seata/seata/issues/4630

### CODE REVIEW

File: core/src/main/java/io/seata/core/exception/AbstractExceptionHandler.java [hunk 1]
```diff
@@ -144,6 +144,9 @@ public <T extends AbstractTransactionRequest, S extends AbstractTransactionRespo
         } catch (RuntimeException rex) {
             LOGGER.error("Catch RuntimeException while do RPC, request: {}", request, rex);
             callback.onException(request, response, rex);
+        } catch (Exception ex) {
```
Comment (slievrly) [inline core/src/main/java/io/seata/core/exception/AbstractExceptionHandler.java:147]: Can `callback.execute` throw other exceptions besides `RuntimeException` and `TransactionException`?

Comment (slievrly) [inline core/src/main/java/io/seata/core/exception/AbstractExceptionHandler.java:147]: Can `callback.execute` throw other exceptions besides `RuntimeException` and `TransactionException`?

Comment (Bughue) [reply core/src/main/java/io/seata/core/exception/AbstractExceptionHandler.java:147]: SQLException in DruidAbstractDataSource

Comment (Bughue) [reply core/src/main/java/io/seata/core/exception/AbstractExceptionHandler.java:147]: Should I just catch the sql exception ？tm and tc log screenshot just added in https://github.com/seata/seata/issues/4630

**Pull request closed without merging**